### PR TITLE
Fix vendor request tests

### PR DESCRIPTION
**Pull request by lannka**
This is to fix the test failures in #18410 caused by stubbing `encodeVars`.   

- It also fixed issue that vars in extraUrlParams were not expanded, see results of `byside` for example.
- For vars with no values assigned, expand it as: `${var}` to `!var` , instead of `$var` (`$` will have URL encoding issue).
- Sorted vendors.

### CODE REVIEW

File: extensions/amp-analytics/0.1/test/test-vendors.js [hunk 1]
```diff
@@ -124,27 +127,17 @@ describes.realWin('amp-analytics', {
...
-            sandbox.stub(variables, 'encodeVars').callsFake(
-                function(name, val) {
-                  val = encodeVars.call(this, name, val);
-                  if (val == '') {
-                    return '$' + name;
+            sandbox.stub(ExpansionOptions.prototype, 'getVar').callsFake(
+                function(name) {
+                  let val = this.vars[name];
+                  if (val == null || val == '') {
+                    val = '!' + name;
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-vendors.js]: emmm I feel `!` is not so intuitive as `$`. We may want to document how to generate the test results to vendors.

File: extensions/amp-analytics/0.1/variables.js [hunk 2]
```diff
@@ -218,23 +230,6 @@ export class VariableService {
     });
   }
 
-  /**
-   * @param {string} unusedName Name of the variable. Only used in tests.
-   * @param {string|?Array<string>} raw The values to URI encode.
-   * @return {string} The encoded value.
-   */
-  encodeVars(unusedName, raw) {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/variables.js]: Awesome!!

File: extensions/amp-analytics/0.1/test/test-vendors.js [hunk 3]
```diff
@@ -164,30 +157,48 @@ describes.realWin('amp-analytics', {
...
+function writeOutput(vendor, name, url) {
+  if (!actualResults[vendor]) {
+    actualResults[vendor] = {};
+  }
+  actualResults[vendor][name] = url;
+  const cnt = Object.keys(AnalyticsConfig[vendor]['requests']).length;
+  AnalyticsConfig[vendor].testCnt = (AnalyticsConfig[vendor].testCnt || 0) + 1;
+  if (cnt == AnalyticsConfig[vendor].testCnt) {
+    delete AnalyticsConfig[vendor];
+    if (Object.keys(AnalyticsConfig).length == 1) {
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/test/test-vendors.js]: this is actually not part of the test. it's try to detect the end of all tests, and print out the expected results, so I can copy paste to `vendor-request.json`

File: extensions/amp-analytics/0.1/test/test-vendors.js (hunk 1 above)
Comment (lannka) [reply extensions/amp-analytics/0.1/test/test-vendors.js]: right, but `$` will be urlencoded. I found `!` is not used in any vendor config, so it's good.

File: extensions/amp-analytics/0.1/vendors.js [hunk 4]
```diff
@@ -163,11 +163,13 @@ export const ANALYTICS_CONFIG = /** @type {!JsonObject} */ ({
     },
   },
   'acquialift': ACQUIALIFT_CONFIG,
+  'adobeanalytics': ADOBEANALYTICS_CONFIG,
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/vendors.js]: sounds good

**Pull request merged**